patches/api/0399-Add-LivingEntity-swingHand-EquipmentSlot-convenience.patch

```diff
@@ -0,0 +1,34 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: SoSeDiK <mrsosedik@gmail.com>
+Date: Tue, 11 Oct 2022 22:35:56 +0300
+Subject: [PATCH] Add LivingEntity#swingHand(EquipmentSlot) convenience method
+
+
+diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
+index 4af5e8d0cba6555f7615e4e809d9aff221c0dc4d..3dcb117664bfd170d739a4ee2af3ad30de298b25 100644
+--- a/src/main/java/org/bukkit/entity/LivingEntity.java
++++ b/src/main/java/org/bukkit/entity/LivingEntity.java
+@@ -985,5 +985,23 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+      * @param hurtDirection hurt direction
+      */
+     void setHurtDirection(float hurtDirection);
++
++    /**
++     * Makes this entity swing their hand.
++     *
++     * <p>This method does nothing if this entity does not
++     * have an animation for swinging their hand.
++     *
++     * @param hand hand to be swung, either {@link org.bukkit.inventory.EquipmentSlot#HAND} or {@link org.bukkit.inventory.EquipmentSlot#OFF_HAND}
++     * @throws IllegalArgumentException if invalid hand is passed
++     */
++    default void swingHand(@NotNull org.bukkit.inventory.EquipmentSlot hand) {
++        com.google.common.base.Preconditions.checkArgument(hand == org.bukkit.inventory.EquipmentSlot.HAND || hand == org.bukkit.inventory.EquipmentSlot.OFF_HAND, "EquipmentSlot was not a hand!");
```

Comment:
Would be nice to include the value of `hand` in the exception message

as far as isHand, it seems reasonable to me. probably do it in a separate PR since with the aforementioned change this one is basically ready

Reply:
Hopefully, this will do :)